Doesn't build here:
-- Build date: 2023-03-11
CMake Error at CMakeLists.txt:97 (message):
  Ragel state machine compiler not found


-- Configuring incomplete, errors occurred!

--- a/components/library/hyperscan/Makefile
+++ b/components/library/hyperscan/Makefile
@@ -1,0 +1,64 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+
+#
+# Copyright (c) 2023, Friedrich Kink
+#
+BUILD_STYLE=	cmake
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		hyperscan
+COMPONENT_VERSION=	5.4.1
+COMPONENT_SUMMARY=	Hyperscan is a high-performance multiple regex matching library.
+COMPONENT_PROJECT_URL=  https://www.hyperscan.io/
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_VERSION).tar.gz
+COMPONENT_ARCHIVE_HASH= sha256:6798202350ecab5ebe5063fbbb6966c33d43197b39ce7ddfbca2e61ac5ecb54a
+COMPONENT_ARCHIVE_URL=  https://github.com/intel/$(COMPONENT_NAME)/archive/refs/tags/v$(COMPONENT_ARCHIVE)
+COMPONENT_CLASSIFICATION=Development/C
+COMPONENT_FMRI=		library/hyperscan
+COMPONENT_CLASSIFICATION=	Development/C
+COMPONENT_LICENSE=	BSD
+COMPONENT_LICENSE_FILE=	LICENSE
+
+include $(WS_MAKE_RULES)/common.mk
+
+PATH=$(PATH.gnu)
+
+# turn on largefile support
+CFLAGS += $(CPP_LARGEFILES)
+
+# to avoid to modern CPU's when build automatically on newer hardware
+CMAKE_OPTIONS+= -DCMAKE_C_FLAGS="-march=corei7"
+CMAKE_OPTIONS+= -DCMAKE_BUILD_TYPE=Release
+CMAKE_OPTIONS+= -DFAT_RUNTIME=OFF
+CMAKE_OPTIONS+= -DBUILD_SHARED_LIBS=ON
+
+TEST_TARGET=       $(NO_TESTS)
+
+# manually added
+REQUIRED_PACKAGES += developer/ragel
+
+# Auto-generated dependencies
+REQUIRED_PACKAGES += $(GCC_RUNTIME_PKG)
+REQUIRED_PACKAGES += $(GXX_RUNTIME_PKG)
+REQUIRED_PACKAGES += system/library
+REQUIRED_PACKAGES += system/library/math

--- a/components/library/hyperscan/hyperscan.p5m
+++ b/components/library/hyperscan/hyperscan.p5m
@@ -1,0 +1,41 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2023 Friedrich Kink
+#
+
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.human-version value=$(HUMAN_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+file path=usr/include/hs/hs.h
+file path=usr/include/hs/hs_common.h
+file path=usr/include/hs/hs_compile.h
+file path=usr/include/hs/hs_runtime.h
+link path=usr/lib/$(MACH64)/libhs.so target=libhs.so.5
+file path=usr/lib/$(MACH64)/libhs.so.$(HUMAN_VERSION)
+link path=usr/lib/$(MACH64)/libhs.so.5 target=libhs.so.$(HUMAN_VERSION)
+link path=usr/lib/$(MACH64)/libhs_runtime.so target=libhs_runtime.so.5
+file path=usr/lib/$(MACH64)/libhs_runtime.so.$(HUMAN_VERSION)
+link path=usr/lib/$(MACH64)/libhs_runtime.so.5 \
+    target=libhs_runtime.so.$(HUMAN_VERSION)
+file path=usr/lib/$(MACH64)/pkgconfig/libhs.pc
+file path=usr/share/doc/hyperscan/examples/README.md
+file path=usr/share/doc/hyperscan/examples/patbench.cc
+file path=usr/share/doc/hyperscan/examples/pcapscan.cc
+file path=usr/share/doc/hyperscan/examples/simplegrep.c

--- a/components/library/hyperscan/manifests/sample-manifest.p5m
+++ b/components/library/hyperscan/manifests/sample-manifest.p5m
@@ -1,0 +1,41 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2023 <contributor>
+#
+
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.human-version value=$(HUMAN_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+file path=usr/include/hs/hs.h
+file path=usr/include/hs/hs_common.h
+file path=usr/include/hs/hs_compile.h
+file path=usr/include/hs/hs_runtime.h
+link path=usr/lib/$(MACH64)/libhs.so target=libhs.so.5
+file path=usr/lib/$(MACH64)/libhs.so.$(HUMAN_VERSION)
+link path=usr/lib/$(MACH64)/libhs.so.5 target=libhs.so.$(HUMAN_VERSION)
+link path=usr/lib/$(MACH64)/libhs_runtime.so target=libhs_runtime.so.5
+file path=usr/lib/$(MACH64)/libhs_runtime.so.$(HUMAN_VERSION)
+link path=usr/lib/$(MACH64)/libhs_runtime.so.5 \
+    target=libhs_runtime.so.$(HUMAN_VERSION)
+file path=usr/lib/$(MACH64)/pkgconfig/libhs.pc
+file path=usr/share/doc/hyperscan/examples/README.md
+file path=usr/share/doc/hyperscan/examples/patbench.cc
+file path=usr/share/doc/hyperscan/examples/pcapscan.cc
+file path=usr/share/doc/hyperscan/examples/simplegrep.c

--- a/components/library/hyperscan/patches/ethernet_h.patch
+++ b/components/library/hyperscan/patches/ethernet_h.patch
@@ -1,0 +1,24 @@
+--- hyperscan-5.4.1/examples/pcapscan.cc	2023-02-26 17:13:53.853385829 +0100
++++ hyperscan-5.4.1/examples/pcapscan.cc.new	2023-02-26 17:57:39.048349531 +0100
+@@ -68,7 +68,8 @@
+ #include <netinet/tcp.h>
+ #include <netinet/udp.h>
+ #include <netinet/ip_icmp.h>
+-#include <net/ethernet.h>
++#include <libnet.h>
++#include <sys/ethernet.h>
+ #include <arpa/inet.h>
+ 
+ #include <pcap.h>
+--- hyperscan-5.4.1/examples/patbench.cc	2022-09-13 14:57:08.000000000 +0200
++++ hyperscan-5.4.1/examples/patbench.cc.new	2023-02-26 18:25:39.051761823 +0100
+@@ -134,7 +134,8 @@
+ #include <netinet/tcp.h>
+ #include <netinet/udp.h>
+ #include <netinet/ip_icmp.h>
+-#include <net/ethernet.h>
++#include <libnet.h>
++#include <sys/ethernet.h>
+ #include <arpa/inet.h>
+ 
+ #include <pcap.h>

--- a/components/library/hyperscan/pkg5
+++ b/components/library/hyperscan/pkg5
@@ -1,0 +1,15 @@
+{
+    "dependencies": [
+        "SUNWcs",
+        "developer/ragel",
+        "shell/ksh93",
+        "system/library",
+        "system/library/g++-10-runtime",
+        "system/library/gcc-10-runtime",
+        "system/library/math"
+    ],
+    "fmris": [
+        "library/hyperscan"
+    ],
+    "name": "hyperscan"
+}

